packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -117,13 +126,16 @@ function resolveCurrentlyRenderingFiber(): Fiber {
   return currentlyRenderingFiber;
 }
 
-export function prepareToUseHooks(
+export function renderWithHooks(
```

Comment:
Why this change of pattern?

Reply:
It made more sense in an earlier iteration of this PR. It's not strictly necessary but the re-render logic is less fragile this way. I can move this to https://github.com/facebook/react/pull/14591 if needed.